pkg/test/framework/components/cluster/version.go

```diff
@@ -0,0 +1,76 @@
+package cluster
+
+import (
+	"strings"
+
+	"istio.io/istio/pilot/pkg/model"
+)
+
+const (
+	// Latest represents a version of Istio running from master
+	Latest = "default"
+)
+
+// Version is an Istio version running within a cluster
+type Version string
+
+// Versions represents a collection of Istio versions running in a cluster
+type Versions []Version
+
+// ToRevision goes from an Istio version to the canonical revision for that version
+func (v Version) ToRevision() string {
+	ver := string(v)
+	if ver == Latest {
+		return Latest
+	}
+	return strings.ReplaceAll(string(v), ".", "-")
+}
+
+func (v Version) Compare(other Version) int {
+	ver := model.ParseIstioVersion(string(v))
```

Comment:
Hmm ... Istio already defines a version structure? Should we just expand it to do what we need here?